app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -139,22 +150,32 @@ class SubscriptionController @Inject()(
       sessionMissingRedirect()
     })
 
-  def redirectSubscriptionResponse(either: Either[SubscriptionReturnedHttpError, (Arn, String)]): Result =
+  private def redirectSubscriptionResponse(either: Either[SubscriptionReturnedHttpError, (Arn, String)])(
+    implicit hc: HeaderCarrier): Future[Result] =
     either match {
       case Right((arn, _)) =>
         mark("Count-Subscription-Complete")
-        Redirect(routes.SubscriptionController.showSubscriptionComplete())
-          .flashing("arn" -> arn.value)
+        redirectUponSuccessfulSubscription(arn)
 
       case Left(SubscriptionReturnedHttpError(CONFLICT)) =>
         mark("Count-Subscription-AlreadySubscribed-APIResponse")
-        Redirect(routes.CheckAgencyController.showAlreadySubscribed())
+        Future.successful(Redirect(routes.CheckAgencyController.showAlreadySubscribed()))
 
       case Left(SubscriptionReturnedHttpError(status)) =>
         mark("Count-Subscription-Failed")
         throw new HttpException("Subscription failed", status)
     }
 
+  private[controllers] def redirectUponSuccessfulSubscription(arn: Arn)(implicit hc: HeaderCarrier) =
```

Comment:
refactor this to what we have talked about, at this point the user would already be subscribed and not be informed about it, and they can drop out of the journey making changes to their account before ever receiving the complete page.

Reply:
Will raise a tech debt ticket for this so we don't forget and can discuss it

Reply:
Captured in APB-2891